tests/test_types.py

```diff
@@ -315,18 +315,47 @@ class Config:
     'field,value,result',
     [
         ('bool_check', True, True),
-        ('bool_check', False, False),
-        ('bool_check', None, False),
-        ('bool_check', '', False),
         ('bool_check', 1, True),
-        ('bool_check', 'TRUE', True),
-        ('bool_check', b'TRUE', True),
+        ('bool_check', 'y', True),
+        ('bool_check', 'Y', True),
+        ('bool_check', 'yes', True),
+        ('bool_check', 'Yes', True),
+        ('bool_check', 'YES', True),
         ('bool_check', 'true', True),
-        ('bool_check', '1', True),
-        ('bool_check', '2', False),
-        ('bool_check', 2, True),
+        ('bool_check', 'True', True),
+        ('bool_check', 'TRUE', True),
         ('bool_check', 'on', True),
-        ('bool_check', 'yes', True),
+        ('bool_check', 'On', True),
+        ('bool_check', 'ON', True),
+        ('bool_check', '1', True),
+        ('bool_check', 't', True),
+        ('bool_check', 'T', True),
+        ('bool_check', b'TRUE', True),
+        ('bool_check', False, False),
+        ('bool_check', 0, False),
+        ('bool_check', 'n', False),
+        ('bool_check', 'N', False),
+        ('bool_check', 'no', False),
+        ('bool_check', 'No', False),
+        ('bool_check', 'NO', False),
+        ('bool_check', 'false', False),
+        ('bool_check', 'False', False),
+        ('bool_check', 'FALSE', False),
+        ('bool_check', 'off', False),
+        ('bool_check', 'Off', False),
+        ('bool_check', 'OFF', False),
+        ('bool_check', '0', False),
+        ('bool_check', 'f', False),
+        ('bool_check', 'F', False),
+        ('bool_check', b'FALSE', False),
+        ('bool_check', None, ValidationError),
+        ('bool_check', '', ValidationError),
+        ('bool_check', [], ValidationError),
+        ('bool_check', {}, ValidationError),
+        ('bool_check', [True], ValidationError),
+        ('bool_check', b'2', ValidationError),
+        ('bool_check', '2', ValidationError),
+        ('bool_check', 2, ValidationError),
```

Comment:
I replaced the test cases with all allowed values excluding bytes versions, along with one bytes version each for True and False. I also added what I figured would be the most common error cases.

Reply:
This is missing the case for a dict of things, and an array of things.

```
bool({}) == False
bool({"test": 1}) == True
bool([]) == False
bool([1]) == True
```

Reply:
I guess for completeness and "tests as a definition of behaviour" we should also include a test for an instance of a class with an explicit `__bool__` method.

Reply:
@samuelcolvin if it has an explicit `__bool__` method, should it be used? Or should it fail validation? I'd expect it to be used in this non-strict case, but just want to confirm first.

Reply:
well that's debatable, I guess personally I think it should be be accepted by I know @cazgp would like it to be stricter.

I guess it doesn't matter that much since nothing coming from "dumb deserialization" eg. JSON will have such a method.

Reply:
@samuelcolvin 
```python
Python 3.7.3
>>> hasattr(2, "__bool__")
True
>>> hasattr(2.5, "__bool__")
True
```
given this, how would you propose to handle the `__bool__` attribute? I could add checks for int / float and handle these cases appropriately, but this has me worried about unexpected edge cases and/or needing to handle a long list of them.

 @cazgp Any thoughts? (I'm assuming you are in favor of dropping `__bool__` support, but figured I'd ask)

Reply:
In that case I think it would be easiest to not include `__bool__` support.

Reply:
Yeah, `__bool__` seems way too open and if it undermines all of the above we should definitely not take it into account!

Reply:
Fine by me, but I propose a sweepstake on how long it will take for someone to request `RelaxedBool` that behaves like python's `bool`.

My guess - within a week of this being released.

Reply:
@samuelcolvin I got in front of that and just added RelaxedBool. Let me know if you want me to remove it